### ISSUE TITLE
`.explain()` on `TreeNode` Models Raises an Exception

### ISSUE THREAD

**Issue by gsnider2195** [nautobot/nautobot#2577]
### Environment
* Nautobot version (Docker tag too if applicable): v1.5.0a0
* Python version: 3.10.7
* Database platform, version: Postgres 13
* Middleware(s):

### Steps to Reproduce
1. From the ORM `Location.objects.all().explain()`

### Expected Behavior
Return a SQL query explanation

### Observed Behavior
Raises an exception

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.SyntaxError: syntax error at or near "EXPLAIN"
LINE 26:     EXPLAIN SELECT (__tree.tree_depth) AS "tree_depth", (__t...
             ^


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/usr/local/lib/python3.10/site-packages/django/db/models/query.py", line 817, in explain
    return self.query.explain(using=self.db, format=format, **options)
  File "/usr/local/lib/python3.10/site-packages/django/db/models/sql/query.py", line 575, in explain
    return '\n'.join(compiler.explain_query())
  File "/usr/local/lib/python3.10/site-packages/django/db/models/sql/compiler.py", line 1228, in explain_query
    result = list(self.execute_sql())
  File "/usr/local/lib/python3.10/site-packages/django/db/models/sql/compiler.py", line 1175, in execute_sql
    cursor.execute(sql, params)
  File "/usr/local/lib/python3.10/site-packages/django/db/backends/utils.py", line 98, in execute
    return super().execute(sql, params)
  File "/usr/local/lib/python3.10/site-packages/cacheops/transaction.py", line 97, in execute
    result = self._no_monkey.execute(self, sql, params)
  File "/usr/local/lib/python3.10/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/usr/local/lib/python3.10/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/usr/local/lib/python3.10/site-packages/django/db/backends/utils.py", line 79, in _execute
    with self.db.wrap_database_errors:
  File "/usr/local/lib/python3.10/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.10/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.ProgrammingError: syntax error at or near "EXPLAIN"
LINE 26:     EXPLAIN SELECT (__tree.tree_depth) AS "tree_depth", (__t...
```

Affects the `random_instance` utility function:
https://github.com/nautobot/nautobot/pull/2514#discussion_r987327071

**Comment by lampwins** [comment]
django-mptt backed models appear to be fine so we might want to explore this as an issue in django-tree-queries per Glenn's comment linked above.

**Comment by bryanculver** [comment]
I opened an issue with `django-tree-queries`: https://github.com/matthiask/django-tree-queries/issues/46

Confirmed this isn't affecting MPTT or non-tree-based models:

```
>>> Device.objects.all().explain()
'Sort  (cost=11.91..12.04 rows=50 width=1508)\n  Sort Key: _name\n  ->  Seq Scan on dcim_device  (cost=0.00..10.50 rows=50 width=1508)'
>>> Region.objects.all().explain()
'Sort  (cost=13.33..13.53 rows=80 width=946)\n  Sort Key: tree_id, lft\n  ->  Seq Scan on dcim_region  (cost=0.00..10.80 rows=80 width=946)'
```

**Comment by bryanculver** [comment]
Can be resolved by updating `django-tree-queries` to 0.12: https://github.com/matthiask/django-tree-queries/blob/main/CHANGELOG.rst#012-2022-11-30

**Comment by bryanculver** [comment]
Formally it seems but [PyPI](https://pypi.org/project/django-tree-queries/0.12.0/), their [setup.cfg](https://github.com/matthiask/django-tree-queries/blob/0.12/setup.cfg#L32), and Poetry don't suggest it actually being dropped. It appears the maintainer has just stopped running CI for it:


```
➜ poetry add "django-tree-queries~0.12"

Updating dependencies
Resolving dependencies... (0.7s)

Writing lock file

Package operations: 0 installs, 1 update, 0 removals

  • Updating django-tree-queries (0.11.0 -> 0.12.0)

nautobot on  develop [$!]
➜ git diff HEAD
```
```diff
diff --git a/poetry.lock b/poetry.lock
index bd4847876..2408d2f10 100644
--- a/poetry.lock
+++ b/poetry.lock
@@ -733,7 +733,7 @@ rest-framework = ["djangorestframework (>=3.0.0)"]

 [[package]]
 name = "django-tree-queries"
-version = "0.11.0"
+version = "0.12.0"
 description = "Tree queries with explicit opt-in, without configurability"
 category = "main"
 optional = false
@@ -2628,7 +2628,7 @@ sso = ["social-auth-core"]
 [metadata]
 lock-version = "1.1"
 python-versions = "^3.7"
-content-hash = "1da5ab83e44d84a421ea00d3a51ae189d10f4442a613a731f57ad8cfa962972c"
+content-hash = "79093bf191a45db186eb37e2285bb1d06566860b14c497d08a8b5fb9a584dbd2"

 [metadata.files]
 amqp = [
@@ -3014,8 +3014,8 @@ django-timezone-field = [
     {file = "django_timezone_field-4.1.2-py3-none-any.whl", hash = "sha256:897c06e40b619cf5731a30d6c156886a7c64cba3a90364832148da7ef32ccf36"},
 ]
 django-tree-queries = [
-    {file = "django_tree_queries-0.11.0-py3-none-any.whl", hash = "sha256:d74fe9e36dc67cb2c6f7df1969ef700627b0c49af2ada15990dd3ae44e4ddb23"},
-    {file = "django_tree_queries-0.11.0.tar.gz", hash = "sha256:768bc75fd5ab617e19bec5c7e207964d7f2f6155f5f3d8c3332b41d9d0e6b436"},
+    {file = "django_tree_queries-0.12.0-py3-none-any.whl", hash = "sha256:fcb6ca3c0d21d0a3576faf846fda7936aa4183b7433d56670af52fc038b72844"},
+    {file = "django_tree_queries-0.12.0.tar.gz", hash = "sha256:25aa63fc0e47177360cedf99245adc10ea247690bd9af32ebf4e44dad15b7c57"},
 ]
 django-webserver = [
     {file = "django-webserver-1.2.0.tar.gz", hash = "sha256:c976979d15b5ff9a212f7904d3b779e22219aebb4857860fcaf20e4e40f1da40"},
diff --git a/pyproject.toml b/pyproject.toml
index 0719d1229..bace1fdaf 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -82,7 +82,7 @@ django-taggit = "~3.0.0"
 # NOTE: django-timezone-field 4.2.x is available but appears to break our initial migrations?
 django-timezone-field = "~4.1.2"
 # Tree database structures based on Common Table Expressions
-django-tree-queries = "~0.11"
+django-tree-queries = "~0.12"
 # Run production webservers such as uWSGI/gunicorn as a Django management command.
 django-webserver = "~1.2.0"
 # REST API framework
```

**Comment by glennmatthews** [comment]
Resolved by #2942.
